datastore/group.go

```diff
@@ -108,3 +111,59 @@ func (db *groupRepo) FetchGroupByID(ctx context.Context,
 
 	return org, err
 }
+
+func (db *groupRepo) DeleteGroup(ctx context.Context, uid string) error {
+
+	update := bson.M{
+		"$set": bson.M{
+			"deleted_at":      primitive.NewDateTimeFromTime(time.Now()),
+			"document_status": convoy.DeletedDocumentStatus,
+		},
+	}
+
+	rollback := bson.M{
+		"$set": bson.M{
+			"deleted_at":      nil,
+			"document_status": convoy.ActiveDocumentStatus,
+		},
+	}
+
+	groupFilter := bson.M{"uid": uid}
+	_, err := db.inner.UpdateOne(ctx, groupFilter, update)
+	if err != nil {
+		return err
+	}
+
+	appFilter := bson.M{"group_id": uid}
+
+	appCollection := db.innerDB.Collection(AppCollections)
```

Comment:
@danvixent preferably, let's create `Delete` methods for applications & events and used their methods respectively vs. writing it here.

Reply:
done, but watch out for a TODO I added

Reply:
Seen. Instead, we should wrap the entire thing in a transaction. But apparently, the current design doesn't expose the datastore to the handlers to start a transaction so another PR.